extensions/wikia/Flags/controllers/FlagsApiController.class.php

```diff
@@ -0,0 +1,336 @@
+<?php
+
+/**
+ * The public interface of the extension and the main entry point for all requests.
+ * It provides a set of CRUD methods to manipulate Flags instances and their types.
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ * @copyright (c) 2015 Wikia, Inc.
+ * @license http://www.gnu.org/copyleft/gpl.html GNU General Public License 2.0 or later
+ */
+
+use Flags\FlagsCache;
+use Flags\Models\Flag;
+use Flags\Models\FlagType;
+
+class FlagsApiController extends WikiaApiController {
+	private
+		$cache,
+		$params,
+		$status = false;
+
+	/**
+	 * Returns a singleton instance of FlagsCache
+	 * @return FlagsCache
+	 */
+	private function getCache() {
```

Comment:
Ok, I understand. The methods have been grouped in terms of a related action, but exposing the `public` methods at the top makes more sense. Thank you!